core/src/test/java/com/capitalone/dashboard/repository/EmbeddedMongoDBRule.java

```diff
@@ -24,89 +24,134 @@
-                new MongodConfigBuilder().version(Version.Main.PRODUCTION)
-                    .net(new Net(port, Network.localhostIsIPv6())).build();
-
-        Command command = Command.MongoD;
-        IRuntimeConfig runtimeConfig =
-                new RuntimeConfigBuilder()
-                    .defaultsWithLogger(command, LOGGER)
-                    .artifactStore(
-                            new ArtifactStoreBuilder().defaults(command).download(
-                                    new DownloadConfigBuilder().defaultsForCommand(command).proxyFactory(new SystemProxy())))
-                    .build();
-
-        MongodStarter runtime = MongodStarter.getInstance(runtimeConfig);
-        mongoExec = runtime.prepare(conf);
-
-        mongoProc = mongoExec.start();
-
-        client = new MongoClient(new ServerAddress(conf.net().getServerAddress(), conf.net().getPort()));
-
-        // set the property for our config...
-        System.setProperty("dbhost", conf.net().getServerAddress().getHostAddress());
-        System.setProperty("dbport", Integer.toString(conf.net().getPort()));
-    }
-
-    @Override
-    public void after() {
-        if (client != null) {
-            client.close();
-            client = null;
-        }
-
-        if (mongoProc != null) {
-            mongoProc.stop();
-            mongoProc = null;
-        }
-        if (mongoExec != null) {
-            mongoExec.stop();
-            mongoExec = null;
-        }
-    }
-
-    public MongoClient client() {
-        return client;
-    }
+	private static final Logger LOGGER = LoggerFactory
+			.getLogger(EmbeddedMongoDBRule.class);
+	private static final String MONGO_PORT_PROP = "MONGO_PORT";
+	private MongodExecutable mongoExec;
+	private MongodProcess mongoProc;
+	private MongoClient client;
+
+	static class SystemProxy implements IProxyFactory {
+		@Override
+		public Proxy createProxy() {
+
+			String proxy = System.getenv("HTTP_PROXY");
+
+			if (proxy == null || proxy.isEmpty()) {
+				proxy = System.getProperty("HTTP_PROXY");
+			}
+			try {
+				URL proxyUrl = new URL(proxy);
+
+				// Case for Proxy authentication required
+				try {
+					StringTokenizer tokenizedUrl = new StringTokenizer(proxyUrl
+							.getUserInfo().toString(), ":");
+					final String authUser = tokenizedUrl.nextToken();
+					final String authPassword = tokenizedUrl.nextToken();
+					
+					if ((proxy != null && !proxy.isEmpty())
+							&& (authUser != null && !authUser.isEmpty())
+							&& (authUser != null && !authPassword.isEmpty())) {
+
+						Authenticator.setDefault(new Authenticator() {
+							public PasswordAuthentication getPasswordAuthentication() {
+								return new PasswordAuthentication(authUser,
+										authPassword.toCharArray());
+							}
+						});
+
+						System.setProperty("http.proxyUser", authUser);
+						System.setProperty("http.proxyPassword", authPassword);
+
+					}
+				} catch (NullPointerException | IllegalArgumentException e) {
```

Comment:
dont catch NPEs, this is one of those code smells..  so revert back to way it was but just see if 	hasMoreTokens() is true before trying to nextToken(), that way it will support if there is a username and no pass cases too.